adoc/deployment-azure.adoc

```diff
@@ -0,0 +1,232 @@
+[NOTE]
+====
+You must have completed <<deployment.preparations>> to proceed.
+====
+
+You will use {tf} to deploy the whole infrastructure described in
+<<architecture-azure>>. Then you will use the `skuba` tool to bootstrap the
+{kube} cluster on top of it.
+
+
+[[architecture-azure]]
+=== AZURE Deployment
+
+The AZURE deployment created by our {tf} template files leads to the
+creation of the infrastructure described in the next paragraphs.
+
+==== Setup service principal for terraform login credential
+
+Following the guide link:https://www.terraform.io/docs/providers/azurerm/guides/service_principal_client_secret.html#creating-a-service-principal-in-the-azure-portal[Creating a Service Principal in the Azure Portal], and set up `ARM_CLIENT_ID`, `ARM_CLIENT_SECRET`, `ARM_SUBSCRIPTION_ID`, `and ARM_TENANT_ID` in `container-openrc.sh`.  
+
+[source,bash]
+----
+export ARM_CLIENT_ID=00000000-0000-0000-0000-000000000000
+export ARM_CLIENT_SECRET=000000000000000000000000000000000
+export ARM_SUBSCRIPTION_ID=00000000-0000-0000-0000-000000000000
+export ARM_TENANT_ID=00000000-0000-0000-0000-000000000000
+----
+
+And sourced before deploying terraform script.:
+
+[source,bash]
+----
+source container-openrc.sh
+----
+
+==== Network
+
+The infrastructure is created inside of a user specified AZURE region.
+The resources are currently all located inside of the user specified availability
+zones. All the nodes are placed inside of the same virtual network, within the same
+subnet.
+
+Worker nodes are never exposed to the public internet. On the opposite
+each master node has a public IP address by default. This allows users to
+connect to them via ssh from their computers.
+
+It's also possible to disable this behavior and make **all** the nodes private.
+This can be done setting the `create_bastionhost` variable to `true`.
+
+When this variable is set all the master nodes cease to have a public IP address.
+An link:https://docs.microsoft.com/en-us/azure/bastion/bastion-overview[Azure Bastion]
+instance is created which becomes the only way to ssh into the cluster.
+
+==== Load Balancer
+
+The {tf} template files take care of creating a
+link:https://azure.amazon.com/elasticloadbalancing/[Classic Load Balancer]
+which exposes the Kubernetes API service deployed on the control plane
+nodes.
+
+The load balancer exposes the following ports:
+
+* `6443`: Kubernetes API server
+* `32000`: Dex (OIDC Connect)
+* `32001`: Gangway (RBAC Authenticate)
+
+==== Accessing the nodes
+
+A default `sles` user is created on each node of the cluster. The user has
+administrator capabilities by using the `sudo` utility.
+
+By default password based authentication is disabled. It's possible to log
+using the ssh key specified via the `admin_ssh_key` variable.
+It's also possible to enable password based authentication by specifying a
+value for the `admin_password` variable. Note well: Azure has some security
+checks in place to avoid the usage of weak passwords.
```

Comment:
```suggestion
By default, password based authentication is disabled. It's possible to log
using the ssh key specified via the `admin_ssh_key` variable.
It's also possible to enable password based authentication by specifying a
value for the `admin_password` variable. 
*Note*: Azure has some security checks in place to avoid the usage of weak passwords.
```